ci/jenkins/stop-stale-jobs.sh

```diff
@@ -0,0 +1,161 @@
+    echo "====== checking for capv stale running jobs on PR ${PULL_REQUEST_ID} ======"
+    capv_jobs="$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json")"
+    length_of_capv_jobs=$(echo "$capv_jobs" | jq "[.[]][10]" | jq length)
+
+    for ((i=0; i<$length_of_capv_jobs; i++)); do
+    
+        capv_job_name=$(echo "$capv_jobs" | jq "[.[]][10][$i].name" | tr -d '"')
+
+        # Job color is the status of the jobs, for example if jobs are in running state then its value will be {"red_anime","blue_anime","aborted_anime".....}
+        # so here "*_anime" means job is in running state.
+        # red_anime: last failed job is in running state.
+        # blue_anime: last successful job is in running state.
+        # aborted_anime: last aborted job is in running state.   
+        job_color=$(echo "$capv_jobs" | jq "[.[]][10][$i].color" | tr -d '"')
+
+        if [[ ${job_color: -5} == "anime" ]]; then
+            last_build_number=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/api/json" | jq "[.[]][10][0].number" | tr -d '"')
+            jenkins_variable="$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json")"
+            pr_id=$(echo "$jenkins_variable" | jq "[.[]][1].ghprbPullId"| tr -d '"')
+
+            if [[ "$PULL_REQUEST_ID" == "$pr_id" ]]; then
+                echo "====== abort running job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+
+                # Need to stop first then clean.
+                curl -XPOST "${JENKINS_URL}job/$capv_job_name/$last_build_number/stop" -u admin:$JENKINS_API_TOKEN
+                
+                # Ensures that the job has been aborted before cleanup.
+                status=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/$last_build_number/api/json" | jq "[.[]][14]" | tr -d '"')
+                if [[ "$status" == "ABORTED" ]]; then
+
+                    CLUSTER_NAME=$(echo "$jenkins_variable" | jq "[.[]][1].BUILD_TAG"| tr -d '"')
+                    ip_of_capv_testbed=$(echo "$jenkins_variable" | jq "[.[]][1].SSH_CONNECTION" | awk '{print $3}')
+
+                    cleanup_running_job "$CLUSTER_NAME" "$ip_of_capv_testbed"
+                else
+                    echo "Job $capv_job_name still running on PR ${PULL_REQUEST_ID}"
+                fi
+            fi
+        fi   
+    done
+}
+
+function stop_all() {
+     # Check all stale waiting jobs on PR.
+      abort_queue_jobs
+
+     # Check all stale running jobs on PR.
+      abort_running_jobs   
+}
+
+# Ensures that the script can be run from anywhere.
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+GIT_CHECKOUT_DIR=${THIS_DIR}/../..
+pushd "$THIS_DIR" > /dev/null
+
+function release_static_ip {
+    CLUSTER_NAME=$1
+    ip_of_capv_testbed=$2
+    echo '=== Releasing IP ==='
+    ssh -o StrictHostKeyChecking=no -i /var/lib/jenkins/.ssh/id_rsa -n jenkins@${ip_of_capv_testbed} "cat "$DEFAULT_WORKDIR/host-local.json" | \
```

Comment:
Improve SSH command readability- indentation and error handling in case it failed to release ip

Reply:
Hi, we have used this cmd `set -eo pipefail` at the top of this file so if any cmd failed during execution then it will exit immediately with non-zero status so in case if ssh cmd failed then it will exit immediately from there.